test/integration/src/test/java/io/pravega/test/integration/ReadWriteTest.java

```diff
@@ -0,0 +1,262 @@
+
+        //wait for writers completion
+        for (i = 0; i < writerList.size(); i++) {
+            log.info("get on writer list {} ", writerList.get(i).get());
+            CompletableFuture.allOf(writerList.get(i));
+        }
+
+        // wait for reads = writes
+        while (NUM_EVENTS != eventsReadFromPravega.size()) {
+            Thread.sleep(5);
+        }
+
+        stopReadFlag.set(true);
+
+        //wait for readers completion
+        for (i = 0; i < readerList.size(); i++) {
+            log.info("get on readerlist {} ", readerList.get(i).get());
+            CompletableFuture.allOf(readerList.get(i));
+        }
+
+        log.info("All writers have stopped. Setting Stop_Read_Flag. Event Written Count:{}, Event Read " +
+                "Count: {}", eventData.get(), eventsReadFromPravega.size());
+        assertEquals(NUM_EVENTS, eventsReadFromPravega.size());
+        assertEquals(NUM_EVENTS, new TreeSet<>(eventsReadFromPravega).size()); //check unique events.
+        //stop reading when no. of reads= no. of writes
+        log.info("test {} succeed", "multiReaderWriterTest");
+
+    }
+
+    private CompletableFuture<Void> startNewWriter(final AtomicLong data,
+                                                   final ClientFactory clientFactory) {
+        return CompletableFuture.runAsync(() -> {
+            final EventStreamWriter<Long> writer = clientFactory.createEventWriter(STREAM_NAME,
+                    new JavaSerializer<Long>(),
+                    EventWriterConfig.builder().build());
+            for (int i = 0; i < 100; i++) {
+                try {
+                    long value = data.incrementAndGet();
+                    log.info("writing event {}", value);
+                    writer.writeEvent(String.valueOf(value), value);
+                    writer.flush();
+                } catch (Throwable e) {
+                    log.warn("test exception writing events: {}", e);
+                    break;
+                }
+            }
+            writer.close();
+
+        });
+    }
+
+    private CompletableFuture<Void> startReader(final String id, final ClientFactory clientFactory, final String
+            readerGroupName, final ConcurrentLinkedQueue<Long> readResult, final AtomicLong writeCount, final
+                                                AtomicLong readCount, final AtomicBoolean exitFlag) {
+        return CompletableFuture.runAsync(() -> {
+            final EventStreamReader<Long> reader = clientFactory.createReader(id,
+                    readerGroupName,
+                    new JavaSerializer<Long>(),
+                    ReaderConfig.builder().build());
+            log.info("exit flag before reading {}", exitFlag.get());
+            log.info("readcount before reading {}", readCount.get());
+            log.info("write count before reading {}", writeCount.get());
+
+            while (!(exitFlag.get() && readCount.get() == writeCount.get())) {
```

Comment:
`exitFlag` is not really required here, why not just have the condition `while (readCount.get() <= writeCount.get())`?